tests/fsharp/single-test.fs

```diff
@@ -161,136 +150,101 @@ let generateProjectArtifacts (pc:ProjectConfiguration) targetFramework configura
+    let propsFileName = Path.Combine(directory, "Directory.Build.props")
+    let overridesFileName = Path.Combine(directory, "Directory.Overrides.targets")
+    let projectFileName = Path.Combine(directory, Path.GetRandomFileName() + ".fsproj")
+    try
+        // Clean up directory
+        Directory.CreateDirectory(directory) |> ignore
+        copyFilesToDest cfg.Directory directory
+        try File.Delete(Path.Combine(directory, "FSharp.Core.dll")) with _ -> ()
+        emitFile targetsFileName targetsBody
+        emitFile overridesFileName overridesBody
+        let propsBody = generateProps testCompilerVersion cfg.BUILD_CONFIG
+        emitFile propsFileName propsBody
+        let projectBody = generateProjectArtifacts pc targetFramework cfg.BUILD_CONFIG
+        emitFile projectFileName projectBody
+        use testOkFile = new FileGuard(Path.Combine(directory, "test.ok"))
+        if outputType = OutputType.Exe then
+            exec { cfg with Directory = directory }  cfg.DotNetExe (sprintf "run -f %s" targetFramework)
+        else
+            exec { cfg with Directory = directory }  cfg.DotNetExe "build /t:RunFSharpScript"
+        testOkFile.CheckExists()
+        result <- true
+    finally
+        if result <> false then
+            Directory.Delete(directory, true)
+        else
+            printfn "Configuration: %s" cfg.Directory
+            printfn "Directory: %s" directory
+            printfn "Filename: %s" projectFileName
 
-#if !FSHARP_SUITE_DRIVES_CORECLR_TESTS
-    | FSC_OPT_PLUS_DEBUG -> executeSingleTestBuildAndRun OutputType.Exe "net40" "net472" true
-    | FSC_OPT_MINUS_DEBUG -> executeSingleTestBuildAndRun OutputType.Exe "net40" "net472" false
-    | FSI_FILE -> executeSingleTestBuildAndRun OutputType.Script "net40" "net472" true
+let singleTestBuildAndRunCore cfg extraRef p =
 
-    | FSI_STDIN -> 
-        use cleanup = (cleanUpFSharpCore cfg)
+    match p with
+    | FSC_CORECLR -> executeSingleTestBuildAndRun cfg OutputType.Exe "coreclr" "netcoreapp2.0" true extraRef
+    | FSI_CORECLR_SCRIPT -> executeSingleTestBuildAndRun cfg OutputType.Script "coreclr" "netcoreapp2.0" true extraRef
+#if INCLUDE_NETFX_TESTS
+    | FSC_NETFX -> executeSingleTestBuildAndRun cfg OutputType.Exe "netfx" "net472" true extraRef
+    | FSC_NETFX_DEBUG -> executeSingleTestBuildAndRun cfg OutputType.Exe "netfx" "net472" false extraRef
+    | FSI_NETFX_SCRIPT -> executeSingleTestBuildAndRun cfg OutputType.Script "netfx" "net472" true extraRef
+
+    | FSI_NETFX_SCRIPT_STDIN -> 
+        use cleanup = cleanUpFSharpCore cfg
         use testOkFile = new FileGuard (getfullpath cfg "test.ok")
-        let sources = extraSources |> List.filter (fileExists cfg)
+        let sources = potentialSources |> List.filter (fileExists cfg)
 
         fsiStdin cfg (sources |> List.rev |> List.head) "" [] //use last file, because `cmd < a.txt b.txt` redirect b.txt only
 
         testOkFile.CheckExists()
 
-    | GENERATED_SIGNATURE -> 
```

Comment:
This got moved to a separate helper in `tests.fs` - there's no need to have it here